Dart/src/com/jetbrains/lang/dart/util/DartResolveUtil.java

```diff
@@ -468,12 +468,14 @@ public static DartReference getLeftReference(@Nullable final PsiElement node) {
     }
     DartReference reference = PsiTreeUtil.getParentOfType(node, DartReference.class, false);
     while (reference != null) {
-      final PsiElement parent = reference.getParent();
+      PsiElement parent = reference.getParent();
       if (parent instanceof DartCascadeReferenceExpression) {
-        DartReference[] references = PsiTreeUtil.getChildrenOfType(parent, DartReference.class);
-        DartReference result = references != null && references.length == 2 ? references[0] : null;
-        // if not starts with node
-        return !PsiTreeUtil.isAncestor(result, node, false) ? result : null;
+        parent = parent.getParent();
+        if (parent instanceof DartValueExpression) {
+          return (DartReference) parent.getFirstChild();
```

Comment:
the same: use DartValueExpression.getExpressionList() and check first entry type